subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -5655,27 +5654,41 @@ int lwm2m_socket_start(struct lwm2m_ctx *client_ctx)
 			/** restore character */
 			client_ctx->desthostname[client_ctx->desthostnamelen] = tmp;
 			if (ret < 0) {
-				LOG_ERR("Failed to set TLS_HOSTNAME option: %d", errno);
-				return -errno;
+				ret = -errno;
+				LOG_ERR("Failed to set TLS_HOSTNAME option: %d", ret);
+				goto error;
```

Comment:
Brace for a conflict with https://github.com/zephyrproject-rtos/zephyr/pull/44289

Reply:
Fixed.